src/inspector/main_thread_interface.cc

```diff
@@ -225,11 +225,20 @@ bool MainThreadInterface::WaitForFrontendEvent() {
   dispatching_messages_ = false;
   if (dispatching_message_queue_.empty()) {
     Mutex::ScopedLock scoped_lock(requests_lock_);
-    while (requests_.empty()) incoming_message_cond_.Wait(scoped_lock);
+    while (!stop_waiting_for_frontend_event_requested_ && requests_.empty()) {
+      incoming_message_cond_.Wait(scoped_lock);
+    }
+    stop_waiting_for_frontend_event_requested_ = false;
   }
   return true;
 }
 
+void MainThreadInterface::StopWaitingForFrontendEvent() {
```

Comment:
Tiny, but maybe call it `WakeUpIfWaintingForFrontendEvent`?

Reply:
It doesn't only wake up the worker thread, but also changes the flag to indicate a request to exit the wait loop. IMHO, the current name is more accurate.